_posts/2017-08-18-gsoc_2017-manohar.markdown

```diff
@@ -1,19 +1,94 @@
 ---
-title: Project Title
-subtitle: Project Subtitle
+title: Building Static analysis Services for Holmes Processing
+subtitle: Creating Template and Documentation
 layout: default
 modal-id: 2
 category: gsoc2017
-date: 2017-01-01
-img: temp.svg
-thumbnail: temp.svg
+date: 2017-08-25
+img: manohar.svg
+thumbnail: manohar.svg
 alt: license-img
-repository: https://github.com/HolmesProcessing/gsoc_
-documentation: 
+repository: https://github.com/HolmesProcessing/Holmes-Totem
+documentation: https://docs.holmesprocessing.com/en/latest/rst/components/holmes-totem/services/index.html
 license: 
 
 ---
 
-##### Overview
+# About Me
 
-Content Goes Here
+My name is Manohar Reddy. Currently studying a 4th year Bachelors at IIIT Vadodara. I am mainly interested in distributed computing and kernel development. Holmes Processing is super fun because everything that I learnt in my course curriculum Algorithms, OS, DBMS, AI etc.. is implimented in here. Throughout this GSoC, I've learned a lot of things about Executable both PE and ELF files. Also about how a process gets loaded into memory.  BThis blog is a summary of tasks that I achieved as a part of Google Summer of Code 2017.
+
+# Overview
+
+This project tries to lower the bar of entry into HolmesProcessing by creating a Template to build a new Totem Service. Throughout this project, I built 2 Totem Static Services and documented about Totem Services and how to create a new Totem Service. I created a Template engine which automatically creates a boilerplate code for a particular type of Service to be written in a particular language.
+
+## Creating a Static Analysis Service
+
+A Totem Static Service is a webserver which does analysis for a short period of time. Currently, a Totem Static Service has 4 files. 
+
+- `HelloREST.Scala` to commuinicate with Totem.
+- `HelloWorld.conf` which contains Service configuration.
+- `Helloworld.{lang}` to create a webserver and logic to analyze input file.
+- `Dockerfile` for containerisation.
+
+All the Services has a lot of common code. I put together all the common code and created a template out of it. Using Template engine I created 2 services and about documentation about Totem Services.
+
+### Template Engine
+
+[Template Engine](https://github.com/HolmesProcessing/Holmes-Toolbox/tree/master/Template-Holmes-Totem-Services) Creates all the essential boilerplate code needed to create a Service. This creates all 4 files needed based on the cofiguration. This creates code by replacing `Helloworld` with the `servicename` provided in configuration file. Now a new user who want to create a Totem Service, can use template engine to create a Service and can directly jump in and start concentrating on service logic. This one currently creates RESTFull type Services. This Template has greater potential when gRPC is implemented where user has to do lot of early configuration before writing with Service logic.
```

Comment:
mis spelling of configuration
new user who want**s**

run this paragraph through spelling and grammar checking